openedx/features/enterprise_support/enrollments/utils.py

```diff
@@ -0,0 +1,99 @@
+"""
+Utils for use in enrollment codebase such as views.
+"""
+import logging
+
+from django.core.exceptions import ObjectDoesNotExist  # lint-amnesty, pylint: disable=wrong-import-order
+from django.db import transaction
+
+from common.djangoapps.student.models import User
+from openedx.core.djangoapps.course_groups.cohorts import CourseUserGroup
+from openedx.core.djangoapps.enrollments import api
+from openedx.core.djangoapps.enrollments.errors import CourseEnrollmentError, CourseEnrollmentExistsError
+from openedx.core.djangoapps.enrollments.utils import add_user_to_course_cohort
+from openedx.core.lib.log_utils import audit_log
+from openedx.features.enterprise_support.enrollments.exceptions import (
+    CourseIdMissingException,
+    UserDoesNotExistException
+)
+
+log = logging.getLogger(__name__)
+
+
+def lms_enroll_user_in_course(username, course_id, mode,
```

Comment:
I could also just have written a similar function in edx-enterprise but keeping it here leaves this open for extension/use in the post handler too and seems like a logical place to put a util meant for use with enterprise use case. But open to ideas